substrate/validator-sets/pallet/src/lib.rs

```diff
@@ -372,6 +381,18 @@ pub mod pallet {
     NonExistentValidator,
   }
 
+  #[pallet::hooks]
+  impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+    fn on_initialize(n: BlockNumberFor<T>) -> Weight {
+      if T::ShouldEndSession::should_end_session(n) {
+        Self::rotate_session();
+        T::BlockWeights::get().max_block
```

Comment:
Why is this max_block? Was pallet-session really that cursed?

Reply:
yea lol. Open to suggestions..

Reply:
Let's add a TODO to set a proper weight later.